### ISSUE TITLE
[BUG] Blank when trying to open Marketplaces

### ISSUE THREAD

**Issue by asrulmunir** [FlowiseAI/Flowise#2821]
**Describe the bug**
Blank page displayed when open Marketplaces on self-hosted Flowise 1.8.4.

**To Reproduce**

1. Click Marketplaces

**Expected behavior**
A list of example flows should appear.

**Setup**

-   Installation: docker
-   Flowise Version: 1.8.4
-   OS: Linux Ubuntu
-   Browser: Safari

**Additional context**
Working fine before upgrade to 1.8.4.

Some of the error:

[Error] TypeError: Spread syntax requires ...iterable not be null or undefined
(anonymous function) — index-DTJK32sP.js:1:7900
Ap — index-HbV-re6j.js:62:24285
al — index-HbV-re6j.js:62:42412
(anonymous function) — index-HbV-re6j.js:62:40729
P — index-HbV-re6j.js:47:1602
ee — index-HbV-re6j.js:47:1971

	(anonymous function) (index-DTJK32sP.js:1:8169)
	Ap (index-HbV-re6j.js:62:24285)
	al (index-HbV-re6j.js:62:42412)
	(anonymous function) (index-HbV-re6j.js:62:40729)
	P (index-HbV-re6j.js:47:1602)
	ee (index-HbV-re6j.js:47:1971)

[Error] TypeError: Spread syntax requires ...iterable not be null or undefined — index-DTJK32sP.js:1:7900
	(anonymous function) (index-DTJK32sP.js:1:8169)
	Ap (index-HbV-re6j.js:62:24285)
	al (index-HbV-re6j.js:62:42412)
	(anonymous function) (index-HbV-re6j.js:62:40729)
	P (index-HbV-re6j.js:47:1602)
	ee (index-HbV-re6j.js:47:1971)

[Error] TypeError: r.framework.map is not a function. (In 'r.framework.map((x,c)=>e.jsx(I,{variant:"outlined",size:"small",label:x,style:{marginRight:3,marginBottom:3}},c))', 'r.framework.map' is undefined)
	T4 (index-HbV-re6j.js:62:41207)
	Ia (index-HbV-re6j.js:62:40281)
	$E (index-HbV-re6j.js:62:35191)
	P (index-HbV-re6j.js:47:1602)
	ee (index-HbV-re6j.js:47:1971)

**Comment by Codelytica** [comment]
Same issue, happened while using Filter option to select from drop sown.

**Comment by carylaw** [comment]
Same

**Comment by HenryHengZJ** [comment]
yep, will be fixed in this PR - https://github.com/FlowiseAI/Flowise/pull/2798

**Comment by HenryHengZJ** [comment]
should be resolved in `v2.0.0`

**Comment by EsqerYasen** [comment]
still happeing in my end:
```shell
npx flowise start


get selectedAddress @ inpage.js:1
get @ inpage.js:1
hasSelectedAddress @ requestProvider.js:2
shouldSelectProvider @ requestProvider.js:2
registerProviders @ requestProvider.js:2
(anonymous) @ requestProvider.js:2
(anonymous) @ requestProvider.js:2
(anonymous) @ requestProvider.js:2
(anonymous) @ requestProvider.js:2
index-CPHl8PIk.js:1  TypeError: t[r].usecases is not iterable (cannot read property undefined)
    at index-CPHl8PIk.js:1:9434
    at Wp (index-BEPqkojn.js:62:24283)
    at fl (index-BEPqkojn.js:62:42409)
    at index-BEPqkojn.js:62:40726
    at P (index-BEPqkojn.js:47:1600)
    at MessagePort.q (index-BEPqkojn.js:47:1970)
(anonymous) @ index-CPHl8PIk.js:1
Wp @ index-BEPqkojn.js:62
fl @ index-BEPqkojn.js:62
(anonymous) @ index-BEPqkojn.js:62
P @ index-BEPqkojn.js:47
q @ index-BEPqkojn.js:47
index-BEPqkojn.js:62  TypeError: t.categories.join is not a function
    at H (index-CPHl8PIk.js:1:7777)
    at Array.filter (<anonymous>)
    at as (index-CPHl8PIk.js:1:13826)
    at my (index-BEPqkojn.js:60:19539)
    at Wv (index-BEPqkojn.js:62:3139)
    at iE (index-BEPqkojn.js:62:44833)
    at nE (index-BEPqkojn.js:62:39790)
    at Z4 (index-BEPqkojn.js:62:39718)
    at fp (index-BEPqkojn.js:62:39570)
    at Qv (index-BEPqkojn.js:62:35934)
Bv @ index-BEPqkojn.js:62
LC.n.callback @ index-BEPqkojn.js:62
yS @ index-BEPqkojn.js:60
DS @ index-BEPqkojn.js:62
ZC @ index-BEPqkojn.js:62
X4 @ index-BEPqkojn.js:62
e5 @ index-BEPqkojn.js:62
$a @ index-BEPqkojn.js:62
JC @ index-BEPqkojn.js:62
P @ index-BEPqkojn.js:47
q @ index-BEPqkojn.js:47
index-BEPqkojn.js:62  Uncaught TypeError: t.categories.join is not a function
    at H (index-CPHl8PIk.js:1:7777)
    at Array.filter (<anonymous>)
    at as (index-CPHl8PIk.js:1:13826)
    at my (index-BEPqkojn.js:60:19539)
    at Wv (index-BEPqkojn.js:62:3139)
    at iE (index-BEPqkojn.js:62:44833)
    at nE (index-BEPqkojn.js:62:39790)
    at Z4 (index-BEPqkojn.js:62:39718)
    at fp (index-BEPqkojn.js:62:39570)
```

**Comment by arvind-iyer** [comment]
@EsqerYasen try to update your flowise version. I had the same issue, npm had installed v1.6.x for some reason. Upgraded to v2.1.5 and it is sorted

**Comment by Apop22** [comment]
I installed and uninstalled Node.js and Flowise in all sorts of combinations of versions. For me the versions of Flowise/Node that work together is the combo of Node v22.13.1 and Flowise 2.2.1.

To reinstall Flowise I used first 

npm cache clean --force

and

npm install -g flowise@2.2.1.

Then, when trying to start flowise at the 3000 port with npx flowise start, I've got a message   listen EADDRINUSE: address already in use :::3000. Next I used  npx kill-port 3000 and it sked me

Need to install the following packages:

kill-port@2.0.1

Ok to proceed? (y)

so I said y

and after that did again

npx flowise start



and Markeplaces are there, working. It took to me numerous trials and a few hours to sort this out.
